src/AWS.Deploy.DocGenerator/Generators/DeploymentSettingsFileGenerator.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using AWS.Deploy.CLI;
+using AWS.Deploy.CLI.Commands;
+using AWS.Deploy.Common.IO;
+using AWS.Deploy.DocGenerator.Utilities;
+using AWS.Deploy.ServerMode.Client;
+using AWS.Deploy.ServerMode.Client.Utilities;
+
+namespace AWS.Deploy.DocGenerator.Generators
+{
+    /// <summary>
+    /// Creates documentation for the deployment settings file that can be used as part of a CI/CD pipeline.
+    /// </summary>
+    public class DeploymentSettingsFileGenerator : IDocGenerator, IDisposable
+    {
+        private readonly IToolInteractiveService _toolInteractiveService;
+        private readonly IFileManager _fileManager;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly ServerModeHttpClient _httpClient;
+
+        public DeploymentSettingsFileGenerator(IToolInteractiveService toolInteractiveService, IFileManager fileManager)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _fileManager = fileManager;
+            _cancellationTokenSource = new CancellationTokenSource();
+            _httpClient = ServerModeHttpClientFactory.ConstructHttpClient(ServerModeUtilities.ResolveDefaultCredentials);
```

Comment:
The `DeploymentSettingsFileGenerator` does not need the `_httpClient` to do its job. All it needs is a list of recommendation objects and a directory path where the generated docs will be stored. Both these arguments can be provided to the `Generate` method. This will make the overall design more unit test friendly.

Reply:
Moved this conversation to Slack